exercises/series/test/test_series.c

```diff
@@ -0,0 +1,62 @@
+#include <stdlib.h>
+#include "vendor/unity.h"
+#include "../src/series.h"
+
+// helper routines to validate test results
+static void testSolution(SeriesResults_t * expectedSolution,
+                         SeriesResults_t * actualSolution)
+{
+   // result count correct
+   TEST_ASSERT_EQUAL(expectedSolution->substringCount,
+                     actualSolution->substringCount);
+
+   // test each substring...
+   for (unsigned int index = 0; index < expectedSolution->substringCount;
+        index++) {
+      TEST_ASSERT_EQUAL_STRING(expectedSolution->substring[index],
+                               actualSolution->substring[index]);
+      free(actualSolution->substring[index]);
+   }
+   free(actualSolution->substring);
```

Comment:
Does this free un-malloced memory when the substring count or the substring length are 0?